UnboxTests.swift

```diff
@@ -572,6 +572,59 @@ class UnboxTests: XCTestCase {
             XCTFail("Unexpected error thrown: \(error)")
         }
     }
+    
+    func testUnboxDictionariesArrayWithObjectsWithFailableProperties() {
+        struct Model: Unboxable {
+            let property1: String
+            let property2: String
+            
+            init(unboxer: Unboxer) {
+                self.property1 = unboxer.unbox("notfailable1")
+                self.property2 = unboxer.unbox("notfailable2")
+            }
+        }
+        
+        let array: [UnboxableDictionary] = [
+            [
+                "notfailable1" : "Hello",
+                "notfailable2" : "Hello"
+            ],
+            [
+                "notfailable1" : "Hello"
+            ]
+        ]
+        
+        let items: [Model] = Unbox(array)
+        XCTAssert(items.count == 1, "Unbox did not return correct number of valid elements")
+        
+        let items2: [Model]? = Unbox(array)
+        XCTAssert(items2 == nil, "Unbox should not return data")
+    }
+    
+    func testUnboxDataArrayWithObjectsWithFailableProperties() {
+        struct Model: Unboxable {
+            let property1: String
+            let property2: String
+            
+            init(unboxer: Unboxer) {
+                self.property1 = unboxer.unbox("notfailable1")
+                self.property2 = unboxer.unbox("notfailable2")
+            }
+        }
+        
+        let json = "[{\"notfailable1\":\"Hello\",\"notfailable2\":\"Hello\"},{\"notfailable1\":\"Hello\"}]"
+        
+        guard let data = json.dataUsingEncoding(NSUTF8StringEncoding) else {
+             XCTFail("Could not create data from a string")
+             return
+        }
+        
+        let items: [Model] = Unbox(data)
+        XCTAssert(items.count == 1, "Unbox did not return correct number of valid elements")
+        
+        let items2: [Model]? = Unbox(data)
```

Comment:
> if my program depends on complete backend data (in the form of an array), then with this change there would be no way to express that (right?)

I understand that when you say "complete backend data" means that you want the complete array, or nil when any of its elements failed.

I think this is possible, let me explain it:

If you unbox to a non optional array:
`let items: [Model] = Unbox(data)`
This enforce to return a non optional array, so it is like saying "I want some data in the array even if the unboxing of some of its elements fail"

On the other hand, If you unbox to an optional array:
`let items2: [Model]? = Unbox(data)`
This is like saying "I want the complete array if all of its elements are properly unboxed, otherwise do not return any array". So this is the method to obtain the complete array. Btw, I think this method was working in the same way before this PR happened, I just added the methods to unbox to non-optional arrays. 

Is this clear now? Or maybe your question was not about this?